sql/core/src/main/scala/org/apache/spark/sql/streaming/progress.scala

```diff
@@ -188,7 +178,8 @@ class SourceProgress protected[sql](
   val latestOffset: String,
   val numInputRows: Long,
   val inputRowsPerSecond: Double,
-  val processedRowsPerSecond: Double) extends Serializable {
+  val processedRowsPerSecond: Double,
+  val metrics: ju.Map[String, String] = Map[String, String]().asJava) extends Serializable {
```

Comment:
Do we need the type def here?

Reply:
Don't see neither an answer nor a change.

Reply:
sorry I thought I replied to this. We can have a type def here, but it's not that necessary. I'm wondering what's the standard here? If we usually create a type def in this case then I can create a type def for this map.

Reply:
When the code can be simplified by removing things then that's the suggested way.

Reply:
I tried to add a type for this Map but find it hard to do so. Because we also use this Map[String, String] in java, if we define this alias in this file, we can't import it into the java ReportSourceMetrics File.

If we want to create an alias in java, by extending a map type, then we need to make it public and add another file, then import them in scala files.

The other option is to only use this alias in the two source metrics files, which it's not that efficient to have a type.

Do you have any suggestion about which way we should go to define this alias?

Reply:
I've had a deeper look and I think it's fine.

Reply:
Thanks!